### PR TITLE
Make notes of the initial research to use Zuul for source-git

### LINKED ISSUE #161: Figure out what would it take to use Zuul for source-git in CentOS Stream
Instead of deploying hardly to run source-git related tasks, we though about adding a new (or multiple) Zuul pipeline step(s) to do source-git related actions.

This would work as an extension of the current Zuul pipelines, the current idea being that the step would create a dist-git repo or SRPM from which the Zuul pipeline run the existing steps (build, tests, so on).

This research is about finding out in more detail what such work would consist of.

### PR DESCRIPTION
**Pull request by csomh**
Thanks for @msrb for providing the links for 'Zuul YAML programming 101'!

Fixes #161.

Co-authored-by: Michal Srb <michal@redhat.com>
Signed-off-by: Hunor Csomortáni <csomh@redhat.com>

### CODE REVIEW

**Comment by csomh** [comment]
@TomasTomecek Is there anything else missing I should look into?

File: source-git-in-zuul/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Some docs: https://gitlab.com/redhat/centos-stream/ci-cd/ci-cd-docs/-/blob/main/assemblies/assembly_test-merge-requests.adoc
+
+Repositories to clone and grep through:
+
+- https://gitlab.com/redhat/centos-stream/ci-cd/zuul/jobs-config
+- https://gitlab.com/redhat/centos-stream/ci-cd/zuul/jobs
+- https://pagure.io/zuul-distro-jobs/tree/master
+
+Matrix channel to reach Zuul mainteiners: https://app.element.io/#/room/#sf-ops:matrix.org
```
Comment (TomasTomecek) [inline source-git-in-zuul/README.md:11]:
```suggestion
Matrix channel to reach Zuul maintainers: https://app.element.io/#/room/#sf-ops:matrix.org
```

Comment (TomasTomecek) [reply source-git-in-zuul/README.md:11]: happy to see this channel is now public

File: source-git-in-zuul/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+In the context of Zuul in CentOS Stream a build is actually a [mock-build
+job]. This one starts with executing the [centpkg-fetch-sources] role, which
+executes `centpkg sources` in the dist-git repository.
+
+So, in order to be able to build and test MRs from a source-git repository:
+
+Have something like a `source-git-mock-build` playbook (executed as part of a
+`source-git-build` project-template), which would execute a
+`source-git-srpm-build` role (to be implemented; should set an `srpm` fact
+with the path of the SRPM) and a `mock-build` role.
```
Comment (TomasTomecek) [inline source-git-in-zuul/README.md:35]: +1 to follow the same scheme

**Review by TomasTomecek — APPROVED** [review]
Well, this looks pretty straightforward and honestly doesn't look like too much work.

**Comment by csomh** [comment]
Thank you for the prompt review @TomasTomecek :medal_sports:

**Comment by csomh** [comment]
> Well, this looks pretty straightforward and honestly doesn't look like too much work.

Welllll, testing this might not be so straightforward. I'll add a note about this, too.

**Pull request merged**